src/eddymotion/utils.py

```diff
@@ -0,0 +1,133 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2022 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Utils to sort the DWI data volume indices"""
+
+from enum import Enum
+
+import numpy as np
+
+
+def linear_action(size, **kwargs):
+    """
+    Sort the DWI data volume indices linearly
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+
+    Examples
+    --------
+    >>> list(linear_action(10))
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+    """
+    return range(size)
```

Comment:
And now, we can make it smarter and calculate size from bvals if they are given instead of size:

```suggestion
    if size is None and 'bvals' in kwargs:
        size = len(kwargs['bvals'])
    if size is None:
       raise TypeError("Cannot build iterator without size") 
    return range(size)
```